amqp-kit/client.go

```diff
@@ -311,3 +321,21 @@ func (c *Client) Close() error {
 
 	return nil
 }
+
+func (c *Client) checkExchange(channel *channel, exchange string) error {
+	c.exchLock.Lock()
```

Comment:
It is better to use `RLock()` here. The `.Lock()` method is useful only for map value write operation.

Reply:
In this function not only read operation... see below

Reply:
maybe use RLock and Lock this